tests/test_templatetags.py

```diff
@@ -72,3 +78,226 @@ def test_render_alias_dont_render_draft_aliases(self):
+        site_id = 1
+
+        with self.login_user_context(self.superuser):
+            # A default category, and a new alias is created for the template tag
+            self.render_template_obj(unlimited_template, {}, self.get_request('/'))
+            self.render_template_obj(site_limited_template, {}, self.get_request('/'))
+
+        alias = AliasModel.objects.filter(static_code="limited_alias_code")
+
+        self.assertEqual(len(alias), 2)
+        self.assertEqual(alias[0].static_code, "limited_alias_code")
+        self.assertEqual(alias[0].site, None)
+        self.assertEqual(alias[1].static_code, "limited_alias_code")
+        self.assertEqual(alias[1].site.pk, site_id)
+
+        # Render both templates again and be sure that the original tags are reused
+        self.render_template_obj(unlimited_template, {}, self.get_request('/'))
+        self.render_template_obj(site_limited_template, {}, self.get_request('/'))
+
+        alias_requery = AliasModel.objects.filter(static_code="limited_alias_code")
+
+        self.assertEqual(alias_requery.count(), 2)
+
+    def test_site_limited_alias_displays_the_correct_contents(self):
+        """
+        The correct contents are shown when viewing the static alias in different sites.
+        """
+        unlimited_template = """{% load djangocms_alias_tags %}{% static_alias "site_limit_alias_code" %}"""  # noqa: E501
+        site_limited_template = """{% load djangocms_alias_tags %}{% static_alias "site_limit_alias_code" site %}"""  # noqa: E501
+        site1 = Site.objects.create(domain='site1.com', name='1')
+        site2 = Site.objects.create(domain='site2.com', name='2')
+
+        category = Category.objects.create(name=DEFAULT_STATIC_ALIAS_CATEGORY_NAME)
+
+        unlimited_alias = self._create_alias(
+            plugins=None, name='test alias', category=category, static_code="site_limit_alias_code", site=None)
+        site_limited_alias = self._create_alias(
+            plugins=None, name='test alias', category=category, static_code="site_limit_alias_code", site=site2)
+
+        add_plugin(
+            unlimited_alias.get_placeholder(self.language),
+            'TextPlugin',
+            language=self.language,
+            body='unlimited text',
+        )
+        add_plugin(
+            site_limited_alias.get_placeholder(self.language),
+            'TextPlugin',
+            language=self.language,
+            body='site limited text',
+        )
+
+        # Should show the contents of the unlimited template
+        with override_settings(SITE_ID=site1.pk):
+            site1_unlimited_preview = self.render_template_obj(unlimited_template, {}, self.get_request('/'))
+            site1_limited_preview = self.render_template_obj(site_limited_template, {}, self.get_request('/'))
+
+        self.assertEqual(site1_unlimited_preview, "unlimited text")
+        # FIXME: The limited preview should fall back if no entries exist for the site?
```

Comment:
Intentionally left as FIXME for this PR?

Reply:
Removed, well spotted